lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -8,6 +10,58 @@
 from ..population.dwelltime import DwelltimeModel
 
 
+def _read_txt(testfile, delimiter):
+    """Read text-based csv file with KymoTrack data
+
+    Parameters
+    ----------
+    delimiter : str
+        Delimiter to use.
+
+    Raises
+    ------
+    IOError
+        If the file is in the wrong format.
+    """
+    if isinstance(testfile, os.PathLike):
+        testfile = os.fspath(testfile)
+
+    if isinstance(testfile, str):
+        with open(testfile, "r") as f:
+            header_lines = [f.readline(), f.readline()]
+    else:
+        # Direct StringIO
+        header_lines = [testfile.readline(), testfile.readline()]
+
+    # from v0.13.0, exported CSV files have an additional header line
+    # with the pylake version and CSV version (starting at 2)
+    version_header = re.search(
+        r"Exported with pylake v([\d\.]*) \| track coordinates v(\d)", header_lines[0]
+    )
+
+    # We had a version header
+    if version_header:
+        header_lines = header_lines[1:]  # Consumed the header line
+        pylake_version = version_header.group(1)
+        csv_version = int(version_header.group(2))
+    else:
+        pylake_version = None
+        csv_version = 1
+
+    data = {}
+    try:
+        raw_data = np.loadtxt(testfile, delimiter=delimiter, unpack=True)
+    except ValueError:
+        raise IOError("Invalid file format!")
+
+    header = header_lines[0].rstrip().split(delimiter)
+    track_idx = raw_data[0, :]
+    for key, col in zip(header, raw_data):
+        data[key] = [col[np.argwhere(track_idx == idx).flatten()] for idx in np.unique(track_idx)]
```

Comment:
could pull the `unique()` out of the comprehension, although I doubt you'd get much if any performance gain

Reply:
Ah yeah, I overlooked that. Amending.